pkg/workloads/manager.go

```diff
@@ -79,24 +79,59 @@ const (
 // characters, hyphens, underscores, and dots.
 var workloadNamePattern = regexp.MustCompile(`^[a-zA-Z0-9._-]+$`)
 
-// NewManager creates a new container manager instance.
+// noopStatusManager is a no-op implementation of StatusManager for fallback cases
+type noopStatusManager struct{}
+
+func (*noopStatusManager) CreateWorkloadStatus(_ context.Context, _ string) error { return nil }
+func (*noopStatusManager) GetWorkloadStatus(_ context.Context, _ string) (*WorkloadStatus, string, error) {
+	return nil, "", nil
+}
+func (*noopStatusManager) SetWorkloadStatus(_ context.Context, _ string, _ WorkloadStatus, _ string) {
+}
+func (*noopStatusManager) DeleteWorkloadStatus(_ context.Context, _ string) error { return nil }
+
+// NewManager creates a new workload manager
 func NewManager(ctx context.Context) (Manager, error) {
+	// Create a manager for config access
+	manager, err := client.NewManager(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client manager: %w", err)
+	}
+
 	runtime, err := ct.NewFactory().Create(ctx)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to create runtime: %w", err)
 	}
 
+	statuses := NewStatusManagerFromRuntime(runtime)
+
 	return &defaultManager{
 		runtime:  runtime,
-		statuses: NewStatusManagerFromRuntime(runtime),
+		statuses: statuses,
+		manager:  manager,
```

Comment:
Two things:

1. We have a number of types named `manager`, including this type. Use a more descriptive field name.
2. It seems like the only thing manager is used for is to create an instance of config. Consider adding config as a struct field instead.